### PR TITLE
Grammar: Add Table Maintenance Statements to MySQL dialect

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

Adds table maintenance statements to MySQL:
https://dev.mysql.com/doc/refman/8.0/en/table-maintenance-statements.html

- CHECK TABLE
- CHECKSUM TABLE
- ANALYZE TABLE
- REPAIR TABLE
- OPTIMIZE TABLE

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Review by tunetheweb — DISMISSED** [review]

**Comment by jpy-git** [comment]
@tunetheweb FYI I moved this to draft as I'm adding a couple more statements to this first 👍

**Comment by jpy-git** [comment]
@tunetheweb it's ready for review now 😄

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 1]
```diff
@@ -1433,3 +1452,149 @@ class HelpStatementSegment(BaseSegment):
...
+        OneOf(
+            "QUICK",
+            "EXTENDED",
+        ),
+    )
+
+
+@mysql_dialect.segment()
+class AnalyzeTableStatementSegment(BaseSegment):
+    """A `ANALYZE TABLE` statement.
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_mysql.py:1506]:
```suggestion
    """An `ANALYZE TABLE` statement.
```

File: test/fixtures/dialects/mysql/analyze_table.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ANALYZE NO_WRITE_TO_BINLOG TABLE some_table;
+ANALYZE NO_WRITE_TO_BINLOG TABLE some_table1, some_table2;
+
+ANALYZE LOCAL TABLE some_table;
+ANALYZE LOCAL TABLE some_table1, some_table2;
+
+ANALYZE TABLE some_table
+UPDATE HISTOGRAM ON some_col;
+ANALYZE TABLE some_table
+UPDATE HISTOGRAM ON some_col1, some_col2;
```
Comment (tunetheweb) [inline test/fixtures/dialects/mysql/analyze_table.sql:13]: I find these statements confusing to grok in a quick glance.

they are two statements over four lines.

I would suggest one if the following:
- collapsing to two lines
- separating with a newline.

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Looks good. However got some minor comments I think you should consider before merging.

File: test/fixtures/dialects/mysql/analyze_table.sql (hunk 2 above)
Comment (tunetheweb) [reply test/fixtures/dialects/mysql/analyze_table.sql:13]: FYI this applies to all test cases in this PR btw, not just the ones highlighted above.

Comment (jpy-git) [reply test/fixtures/dialects/mysql/analyze_table.sql]: fixed this now 👍

**Review by tunetheweb — APPROVED** [review]
Thanks for addressing!

**Pull request merged**